libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -29,130 +35,156 @@ class MapboxRouteLineView(var options: MapboxRouteLineOptions) {
-            style,
-            RouteConstants.ALTERNATIVE_ROUTE1_SOURCE_ID,
-            routeDrawData.getAlternativeRoute1Source()
-        )
-        updateSource(
-            style,
-            RouteConstants.ALTERNATIVE_ROUTE2_SOURCE_ID,
-            routeDrawData.getAlternativeRoute2Source()
-        )
-        updateLineGradient(
-            style,
-            RouteLayerConstants.ALTERNATIVE_ROUTE1_TRAFFIC_LAYER_ID,
-            routeDrawData.getAlternativeRoute1TrafficExpression()
-        )
-        updateLineGradient(
-            style,
-            RouteLayerConstants.ALTERNATIVE_ROUTE2_TRAFFIC_LAYER_ID,
-            routeDrawData.getAlternativeRoute2TrafficExpression()
-        )
-        updateSource(
-            style,
-            RouteConstants.WAYPOINT_SOURCE_ID,
-            routeDrawData.getOriginAndDestinationPointsSource()
-        )
+        when (routeDrawData) {
+            is Expected.Success -> {
+                updateLineGradient(
+                    style,
+                    RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID,
+                    routeDrawData.value.getTrafficLineExpression()
+                )
+                updateLineGradient(
+                    style,
+                    RouteLayerConstants.PRIMARY_ROUTE_LAYER_ID,
+                    routeDrawData.value.getRouteLineExpression()
+                )
+                updateLineGradient(
+                    style,
+                    RouteLayerConstants.PRIMARY_ROUTE_CASING_LAYER_ID,
+                    routeDrawData.value.getCasingLineExpression()
+                )
+                updateSource(
+                    style,
+                    RouteConstants.PRIMARY_ROUTE_SOURCE_ID,
+                    routeDrawData.value.getPrimaryRouteSource()
+                )
+                updateSource(
+                    style,
+                    RouteConstants.ALTERNATIVE_ROUTE1_SOURCE_ID,
+                    routeDrawData.value.getAlternativeRoute1Source()
+                )
+                updateSource(
+                    style,
+                    RouteConstants.ALTERNATIVE_ROUTE2_SOURCE_ID,
+                    routeDrawData.value.getAlternativeRoute2Source()
+                )
+                updateLineGradient(
+                    style,
+                    RouteLayerConstants.ALTERNATIVE_ROUTE1_TRAFFIC_LAYER_ID,
+                    routeDrawData.value.getAlternativeRoute1TrafficExpression()
+                )
+                updateLineGradient(
+                    style,
+                    RouteLayerConstants.ALTERNATIVE_ROUTE2_TRAFFIC_LAYER_ID,
+                    routeDrawData.value.getAlternativeRoute2TrafficExpression()
+                )
+                updateSource(
+                    style,
+                    RouteConstants.WAYPOINT_SOURCE_ID,
+                    routeDrawData.value.getOriginAndDestinationPointsSource()
+                )
+            }
+            is Expected.Failure -> { }
```

Comment:
Should we clear the sources here?

Reply:
I think the view class would be making too many assumptions in this case.  The API should make decisions about the line mutations IMO.